File: SDKs/plaid.com/2020-09-14_1.20.6/python/src/sdk/models/shared/investmentsholdingsgetrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import investmentholdingsgetrequestoptions as shared_investmentholdingsgetrequestoptions


@dataclass_json
@dataclasses.dataclass
class InvestmentsHoldingsGetRequest:
    r"""InvestmentsHoldingsGetRequest
    InvestmentsHoldingsGetRequest defines the request schema for `/investments/holdings/get`
    """
    
    access_token: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('access_token') }})
    client_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('client_id') }})
    options: Optional[shared_investmentholdingsgetrequestoptions.InvestmentHoldingsGetRequestOptions] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('options') }})
    secret: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('secret') }})
    

File: SDKs/plaid.com/2020-09-14_1.20.6/java/lib/src/main/java/openapisdk/models/shared/InvestmentsHoldingsGetRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * InvestmentsHoldingsGetRequest
 * InvestmentsHoldingsGetRequest defines the request schema for `/investments/holdings/get`
**/
public class InvestmentsHoldingsGetRequest {
    @JsonProperty("access_token")
    public String accessToken;
    public InvestmentsHoldingsGetRequest withAccessToken(String accessToken) {
        this.accessToken = accessToken;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("client_id")
    public String clientId;
    public InvestmentsHoldingsGetRequest withClientId(String clientId) {
        this.clientId = clientId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("options")
    public InvestmentHoldingsGetRequestOptions options;
    public InvestmentsHoldingsGetRequest withOptions(InvestmentHoldingsGetRequestOptions options) {
        this.options = options;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("secret")
    public String secret;
    public InvestmentsHoldingsGetRequest withSecret(String secret) {
        this.secret = secret;
        return this;
    }
}